Frameworks/Foundation/_NSInvocation.arm.asm

```diff
@@ -84,4 +84,196 @@
     DCB (%B116-%B100)/2 ; RETURN_TYPE_STRUCT
 |_CallFrameInternal| ENDP
 
+    EXTERN _NSInvocation_ForwardFrame
+    GLOBAL _NSInvocation_ForwardingBridgeNoStret
+    GLOBAL _NSInvocation_ForwardingBridge
+    ROUT
+    ALIGN
+    ; r0 = self
+    ; r1 = _cmd
+    ; Saves the first four registers and the VFP argument registers in the ARM call frame format
+    ; expected by NSInvocation. Shifts r0 and r1 up to r1 and r2 to make space for the struct return
+    ; parameter. r0 is filled with nullptr, as there is no struct return.
+|_NSInvocation_ForwardingBridgeNoStret|
+    ;;; This code must match the layout of _NSInvocation_ForwardingBridgeNoStret.prolog.xdata below.
+    ; The prolog is duplicated here so that the exception records can point to one contiguous (branch-free) block
+    ; of instructions for virtual unwinding.
+    PUSH {r0-r3}
+    VSTMDB sp!, {d0-d7}
+    MOV r3, sp
+    PUSH {r4, fp, lr}
+    SUB sp, sp, #12
+|_NSInvocation_ForwardingBridgeNoStret.prolog.end|
+
+    MOV r2, r1 ; sel
+    MOV r1, r0 ; self
+    MOV r0, #0 ; stret
+
+    ; Skip to the body of _NSInvocation_ForwardingBridgeNoStret
+    B %F0
+
+    ; Second entrypoint to this function; called when there is a struct return pointer.
+    ; r0 = struct return pointer
+    ; r1 = self
+    ; r2 = _cmd
+|_NSInvocation_ForwardingBridge|
+    ;;; This code must match the layout of _NSInvocation_ForwardingBridge.prolog.xdata below.
```

Comment:
>;;; This code must match the layout of _NSInvocation_ForwardingBridge.prolog.xdata below. [](start = 4, length = 89)

Is it possible to use the macros from kxarm.h here instead of building your own xdata? PROLOG_PUSH, etc.?